app/src/main/java/com/project/myproject/ui/fragments/RegisterFragment.kt

```diff
@@ -0,0 +1,300 @@
+                        regPasswordLayout.error = getString(R.string.error_password_symbols)
+                    }
+
+                    else -> {
+                        regPasswordLayout.error = null
+                    }
+                }
+            }
+        })
+    }
+
+    private fun setupEmailValidation() {
+        regEmailEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                // Not used
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                // Not used
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                val email = s.toString()
+
+                if (email.isNotEmpty() && !Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
+                    regEmailLayout.error = getString(R.string.error_invalid_email)
+                } else {
+                    regEmailLayout.error = null
+                }
+            }
+        })
+    }
+
+    private fun initializeRegisterButtonListeners() {
+        val registerButton = binding.btnRegisterButton
+
+        registerButton.setOnClickListener {
+            val email = regEmailEditText.text
+            val password = regPasswordEditText.text
+
+            if (regEmailLayout.error == null && regPasswordLayout.error == null
+                && !email.isNullOrBlank() && !password.isNullOrBlank()) {
+
+                viewModel.registerUser(email.toString(), password.toString())
+            }
+
+            if (email.isNullOrBlank()) {
+                regEmailLayout.error = getString(R.string.error_email_required)
+            }
+
+            if (password.isNullOrBlank()) {
+                regPasswordLayout.error = getString(R.string.error_password_required)
+            }
+        }
+    }
+
+    private fun hideAllViewsExceptBackground() {
+        val mainLayout: ConstraintLayout = binding.clMain
+
+        for (i in 0 until mainLayout.childCount) {
+            val view: View = mainLayout.getChildAt(i)
+
+            view.visibility = View.INVISIBLE
+        }
+        binding.registerBackground.visibility = View.VISIBLE
+    }
+
+    private fun fadeAllViewsExceptBackground() {
+        try {
+            val mainLayout: ConstraintLayout = binding.clMain
+
+            for (i in 0 until mainLayout.childCount) {
+                val view: View = mainLayout.getChildAt(i)
+
+                if (binding.registerBackground == view) {
+                    continue
+                }
+
+                view.visibility = View.VISIBLE
+
+                val fadeInAnimation = AnimationUtils.loadAnimation(context, androidx.appcompat.R.anim.abc_fade_in)
+                view.startAnimation(fadeInAnimation)
+            }
+        } catch (e: NullPointerException) {
```

Comment:
add checks for null, do not try catch the code, especially when the catch block is empty